spec/controllers/oidc_controller_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "helpers/webmock_helper"
+
+# rubocop:disable ExampleLength
+RSpec.describe OidcController, type: :controller do
+  let(:external_fqdn_pillar) { create(:external_fqdn_pillar) }
+
+  describe "GET /oidc" do
+    it "returns a 302 if the orchestration didn't yet finish" do
+      VCR.use_cassette("kubeconfig/cluster_not_ready", record: :none) do
+        setup_undone
+
+        get :index
+
+        expect(response.status).to eq 302
+        expect(response.header["Location"]).to eq "http://test.host/"
+      end
+    end
+
+    it "redirects to Dex if orchestration is done" do
+      # VCR can do the salt stuff, we'll just mock Dex
+      VCR.use_cassette("kubeconfig/cluster_ready", record: :none) do
+        setup_done
+        external_fqdn_pillar
+
+        mock_json :get,
+          "https://api.example.com:32000/.well-known/openid-configuration",
+          "oidc/openid-configuration" do
+          get :index
+
+          expect(response.status).to eq 302
+          expect(response.header["Location"]).to include "/auth"
+        end
+      end
+    end
+
+    it "generates a kubeconfig if auth succeeds" do
+      # VCR can do the salt stuff, we'll just mock Dex
+      VCR.use_cassette("kubeconfig/cluster_ready2", record: :none) do
+        setup_done
+        external_fqdn_pillar
+
+        mock_json :get, "https://api.example.com:32000/.well-known/openid-configuration",
```

Comment:
Why don't use `VCR` instead?

Reply:
I spent three days trying to use VCR with no luck. The OIDC handshake is fairly complex - the URLs have to match, the nonce has to match, etc. It was much easier to perform a handshake using the CLI and record the HTTP traffic and use that to form these tests.